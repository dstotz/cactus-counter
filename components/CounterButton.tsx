import { Button, Pressable, StyleSheet, Text, View } from "react-native";

type CounterButtonProps = {
  title: string;
  numCounters: number;
  onPress: (newCounters: number) => void;
};

export const CounterButton = ({
  title,
  numCounters,
  onPress,
}: CounterButtonProps) => {
  return (
    <View style={styles.container}>
      <Pressable
        style={({ pressed }) => [
          {
            backgroundColor: pressed ? "rgb(210, 230, 255)" : "lightgray",
          },
          styles.button,
        ]}
        onPress={() => onPress(numCounters)}
      >
        <Text style={styles.text}>{title}</Text>
      </Pressable>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
  },
  text: {
    textAlign: "center",
    fontSize: 16,
    color: "black",
  },
  button: {
    borderRadius: 8,
    padding: 6,
    width: 50,
  },
});
