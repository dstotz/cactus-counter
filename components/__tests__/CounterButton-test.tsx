import * as React from "react";
import { render } from "@testing-library/react-native";

import { CounterButton } from "../CounterButton";

it(`renders correctly`, () => {
  const tree = render(
    <CounterButton title="+1" numCounters={1} onPress={() => {}} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
