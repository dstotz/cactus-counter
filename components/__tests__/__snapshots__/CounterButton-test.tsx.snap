// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "backgroundColor": "lightgray",
        },
        {
          "borderRadius": 8,
          "padding": 6,
          "width": 50,
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      +1
    </Text>
  </View>
</View>
`;
